rope/base/pyobjects.py

```diff
@@ -63,58 +61,43 @@ def __iter__(self):
         """The same as ``iter(self.get_attributes())``"""
         return iter(self.get_attributes())
 
-    _types = None
-    _unknown = None
 
-    @staticmethod
-    def _get_base_type(name):
-        if PyObject._types is None:
-            PyObject._types = {}
-            base_type = PyObject(None)
-            PyObject._types["Type"] = base_type
-            PyObject._types["Module"] = PyObject(base_type)
-            PyObject._types["Function"] = PyObject(base_type)
-            PyObject._types["Unknown"] = PyObject(base_type)
-        return PyObject._types[name]
+_base_types = None
 
 
-def get_base_type(name):
+def get_base_type(name: str) -> PyObject:
     """Return the base type with name `name`.
 
-    The base types are 'Type', 'Function', 'Module' and 'Unknown'.  It
-    was used to check the type of a `PyObject` but currently its use
-    is discouraged.  Use classes defined in this module instead.
-    For example instead of
-    ``pyobject.get_type() == get_base_type('Function')`` use
-    ``isinstance(pyobject, AbstractFunction)``.
-
-    You can use `AbstractClass` for classes, `AbstractFunction` for
-    functions, and `AbstractModule` for modules.  You can also use
-    `PyFunction` and `PyClass` for testing if an object is
-    defined somewhere and rope can access its source.  These classes
-    provide more methods.
+    Use:        ``isinstance(pyobject, AbstractFunction)``
+    instead of: ``pyobject.get_type() == get_base_type('Function')``.
 
     """
-    return PyObject._get_base_type(name)
+    global _base_types
+    if _base_types is None:
```

Comment:
Rather than initializing `base_types` in `get_base_type()`, why not just initialize it in global scope?

You can't do that with `PyObject._types`, because at that point `PyObject` class wasn't defined yet; but if you're moving `PyObject._types` to the global `base_types`, then there wouldn't be that problem, wouldn't it?